patches/server/0968-New-event-verbose-logging-command.patch

```diff
@@ -202,3 +178,75 @@ index 7ce9ebba8ce304d1f3f21d4f15ee5f3560d7700b..dfb91cb8e19e0159e0eba9367ca73961
  
          HandlerList handlers = event.getHandlers();
          RegisteredListener[] listeners = handlers.getRegisteredListeners();
+diff --git a/src/main/java/io/papermc/paper/util/misc/EventWrapper.java b/src/main/java/io/papermc/paper/util/misc/EventWrapper.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..63c022811e3b409ddbb14bd84dfbdc669121ea1c
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/util/misc/EventWrapper.java
+@@ -0,0 +1,66 @@
++package io.papermc.paper.util.misc;
++
++import com.google.gson.JsonObject;
++import net.minecraft.server.MinecraftServer;
++import org.bukkit.event.Event;
++import org.checkerframework.checker.nullness.qual.Nullable;
++
++import java.lang.reflect.Field;
++import java.lang.reflect.Modifier;
++import java.time.LocalDateTime;
++import java.time.format.DateTimeFormatter;
++import java.util.ArrayList;
++import java.util.Arrays;
++import java.util.List;
++
++/**
++ * A simple event wrapper with a timestamp to allow for QoL JSon serialization
++ */
++public class EventWrapper {
++
++    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd_HH.mm.ss");
++    private final String timestamp;
++    private final Event event;
++
++    public EventWrapper(Event event) {
++        this.timestamp = FORMATTER.format(LocalDateTime.now());
++        this.event = event;
++    }
++
++    /**
++     * Properly serialize our Wrapper into a JsonObject, so we can log it properly
++     *
++     * @return Null if {@link #shouldSerialize(List)} returns false, otherwise it's your object
++     */
++    @Nullable
++    public JsonObject serialize() {
++        JsonObject parent = new JsonObject();
++        parent.addProperty("timestamp", timestamp);
++        JsonObject obj = new JsonObject();
++        obj.addProperty("event", event.getEventName());
++        try {
++            List<Field> fields = new ArrayList<>(Arrays.asList(event.getClass().getDeclaredFields()));
++            if (!shouldSerialize(fields)) return null; //check current event class fields if we should serialize
++            Class<?> clazz = event.getClass().getSuperclass();
++            while (clazz != null) {
++                fields.addAll(Arrays.asList(clazz.getDeclaredFields()));
++                clazz = clazz.getSuperclass();
++            }
```

Comment:
The `shouldSerialize` logic is only applied to the fields on the event class, not all fields in the event type hierarchy. You should collect all the fields while filtering out specific ones.

Also, your shouldSerialize does seem to be incorrect on its face. If *any* of the fields shouldn't serialized, you skip the whole event.

Reply:
> EDIT: Another thing, is I feel some events should be excluded by default. Like the tick events, ones that are just spammed a ton. Can have something that does include those, but most of the time they are just gonna clog up the system. We can also exclude events who don't have "changeable" values.

shouldSerialize was intentionally implemented because of that ^, figured it'd just be a redundant computation in watch command. Only was checking the classes fields for events like `VehicleUpdateEvent`, because I'd assume the inheritance would be irrelevant. 
Will make it global and check all fields with proper logic either way in a bit

Reply:
I guess that kind of only applies to the verbose functionality that logs the event after every listener is called.